site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java

```diff
@@ -192,15 +204,46 @@ private HstQuery buildQuery(HstRequest request) {
             default:
                 log.error("Unknown sort mode: " + sortParam);
                 break;
-        }
+        }    
 
         return constructQuery(queryBuilder, searchStringConstraint);
     }
 
+    private HippoBean[] deriveScope(HstRequest request) {
+        SearchArea areaParam = getAreaOption(request);
+
+        final HippoBean base = RequestContextProvider.get().getSiteContentBaseBean();
+        final HippoBean newsFolder = base.getBean(FOLDER_NEWS);
+        final HippoBean publicationsFolder = base.getBean(FOLDER_PUBLICATIONS);
+        final HippoBean servicesFolder = base.getBean(FOLDER_SERVICES);
+        final HippoBean dataAndInformationFolder = base.getBean(FOLDER_DATAANDINFORMATION);
+        final HippoBean nationalIndicatorLibraryFolder = base.getBean(FOLDER_NIL);
+
+        List<HippoBean> scopeBeans = new ArrayList<HippoBean>();
+
+        if (areaParam.equals(SearchArea.NEWS)) {
+            scopeBeans.add(newsFolder);
+        } else if (areaParam.equals(SearchArea.DATA)) {
+            scopeBeans.add(publicationsFolder);
+            scopeBeans.add(nationalIndicatorLibraryFolder);
+            scopeBeans.add(dataAndInformationFolder);
+        } else if (areaParam.equals(SearchArea.SERVICES)) {
+            scopeBeans.add(servicesFolder);
+        } else { //default/all
+            scopeBeans.add(request.getRequestContext().getSiteContentBaseBean());
+        }
+
+        return scopeBeans.toArray(new HippoBean[scopeBeans.size()]);
+    }
+
     private String getSortOption(HstRequest request) {
         return Optional.ofNullable(getAnyParameter(request, REQUEST_PARAM_SORT)).orElse(SORT_DEFAULT);
     }
 
+    private SearchArea getAreaOption(HstRequest request) {
+        return SearchArea.valueOf(Optional.ofNullable(getAnyParameter(request, REQUEST_PARAM_AREA)).orElse(AREA_DEFAULT).toUpperCase());
```

Comment:
I'd use AREA_DEFAULT as the enum itself rather than converting it to a string and then back to an enum.

```
String area = getAnyParameter(request, REQUEST_PARAM_AREA)

return area == null ? AREA_DEFAULT : SearchArea.valueOf(area)
```

Reply:
Good call